Sources/Datadog/Datadog.swift

```diff
@@ -51,6 +51,10 @@ public class Datadog {
             try initializeOrThrow(
                 configuration: try FeaturesConfiguration(configuration: configuration, appContext: appContext)
             )
+            // TODO: RUMM-511 remove this warning
+            #if targetEnvironment(macCatalyst)
+            userLogger.warn("⚠️ Catalyst is not officially supported by Datadog SDK: use at your own risk!")
```

Comment:
1/. We can't use `userLogger` during the SDK initialization. This logger is only functional if user sets `Datadog.verbosity`, which should happen after `Datadog.initialize()` returns. Instead, we should use `consolePrint()` directly.

2/ I think _"some features will not work correctly"_ sounds better than _"use at your own risk!"_ which suggests damage 🙂, WDYT?

Reply:
i used `userLogger` so that the user could suppress warning (@iGranDav 's idea) by changing `verbosityLevel` but you are right 👍 
can we let the user do so with `consolePrint`?

changing to "Some features may not be functional"? 
eg: carrier info is disabled, some other features may not work

Reply:
> can we let the user do so with `consolePrint`?

No, and it's by design. Depending on the message severity we can either pick:
* `userLogger` - for "warning" purpose which can be suppressed,
* `consolePrint` - for "error" purpose which cannot be suppressed.

Reply:
`userLogger` seems like a better fit as this is a warning though 🤔 
how about that?
```swift
/// Default is `nil` in supported platforms and `.warn` in non-supported platforms
public static var verbosityLevel: LogLevel? = {
  #if target(Catalyst) return .warn
  #else return nil
  #endif
}
```
otherwise i'll leave it without any suppression option.?

Reply:
Current flow is plain simple: `userLogger` considers the `.verbosityLevel`, `consolePrint()` does not. I see no reason to make it more complex in the implementation and tests. Even if it's a "warning", not "error" (debatable), this PR is only a quick win o make the SDK compile on the platform which we do not support.

Reply:
that popped up in the Slack discussion (i've just cc'ed you)

okay 👌 changing to `consolePrint()` ✅